model/infinispan/src/main/java/org/keycloak/models/sessions/infinispan/remotestore/RemoteCacheSessionsLoader.java

```diff
@@ -79,23 +81,36 @@ public WorkerResult loadSessions(KeycloakSession session, RemoteCacheSessionsLoa
         RemoteCache<?, ?> remoteCache = getRemoteCache(session);
 
         int countLoaded = 0;
-        try (CloseableIterator<Map.Entry<Object, Object>> it = remoteCache.retrieveEntries(null, loaderContext.getSessionsPerSegment())) {
+        try (CloseableIterator<Map.Entry<Object, MetadataValue<Object>>> it = remoteCache.retrieveEntriesWithMetadata(null, loaderContext.getSessionsPerSegment())) {
             Map<Object, Object> toInsert = new HashMap<>(loaderContext.getSessionsPerSegment());
             int count = 0;
+            int maxLifespan = -1;
+            int maxIdle = -1;
             while (it.hasNext()) {
-                Map.Entry<?,?> entry = it.next();
-                toInsert.put(entry.getKey(), entry.getValue());
+                Map.Entry<Object, MetadataValue<Object>> entry = it.next();
+                if (entry.getValue().getLifespan() > 0) {
```

Comment:
If the `entry.getValue().getLifespan() < 0` (we skip this if) we still create the entry with some lifespan encountered based on the other entries in the current batch. Should we handle these differently? Maybe create a new `toInsertImmortal`?